dbms/src/Interpreters/Join.cpp

```diff
@@ -1720,22 +1713,31 @@ void Join::waitUntilAllProbeFinished() const
         throw Exception(error_message);
 }
 
-bool Join::isAllProbeFinished() const
+bool Join::quickCheckProbeFinished() const
 {
-    std::lock_guard lock(build_probe_mutex);
-    return probe_finished || meet_error || skip_wait;
+    return probe_finished;
```

Comment:
There is no need to check `meet_error` and `skip_wait` here, because when an error occurs, `exec_context.isCancelled()` is true, and all tasks will end.